dubbo-filter/dubbo-filter-cache/src/main/java/org/apache/dubbo/cache/support/AbstractCacheFactory.java

```diff
@@ -55,15 +58,29 @@ public Cache getCache(URL url, Invocation invocation) {
         url = url.addParameter(METHOD_KEY, invocation.getMethodName());

```

Comment:
You'd better not overwrite the value of `url`

Reply:
I did not modify this piece of code. Can you describe your opinion in detail?

Reply:
The `url` 's value is changed
```java
public Cache getCache(URL url, Invocation invocation) {
    url = url.addParameter(METHOD_KEY, invocation.getMethodName());
    String key = url.toFullString();
    Cache cache = caches.get(key);
}
```

The `url` 's value is unchanged
```java
public Cache getCache(URL url, Invocation invocation) {
    String key = url.addParameter(METHOD_KEY, invocation.getMethodName()).toFullString();
    Cache cache = caches.get(key);
}
```

I know you don't modify this piece of code, but I think it's unreasonable to change the value of an input parameter if the caller don't know that.

Also, I know that if you do it the way I said, you might need to change a lot of code (All the places where `METHOD_KEY` is used may need to be modified)

Reply:
The URL is designed to be immutable. After adding parameters, if the url is not accepted, the new url cannot be obtained. If the METHOD_KEY parameter is added elsewhere, this parameter may not be needed in that place. Do you have any good suggestions?

Reply:
This is indeed a complicated problem. You will change a lot of code if you want to solve this problem completely.

You can refer #7641

For this issue, your solution may be the simplest and least risky

Reply:
Good suggestion. 

I think we should fix this problem first. The problem of url reconstruction will be dealt with later. What do you think so?

Reply:
Yes, you're right!